google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/DoManagedVmDeployment.java

```diff
@@ -73,30 +94,40 @@ public void run() {
           loggingHandler,
           "Created temporary staging directory: " + stagingDirectory.getAbsolutePath());
     } catch (IOException e) {
-      throw new RuntimeException(e);
+      logger.error(e);
+      callback.errorOccurred(
+          GctBundle.message("appengine.deployment.error.creating.staging.directory"));
+      return;
     }
 
-    File stagedArtifactPath =
-        copyFile(stagingDirectory, "target" + artifactType, deploymentArtifactPath);
-    stagedArtifactPath.setReadable(true /* readable */, false /* ownerOnly */);
+    try {
+      File stagedArtifactPath =
+          copyFile(stagingDirectory, "target" + artifactType, deploymentArtifactPath);
+      stagedArtifactPath.setReadable(true /* readable */, false /* ownerOnly */);
 
-    copyFile(stagingDirectory, "app.yaml", this.appYamlPath);
-    copyFile(stagingDirectory, "Dockerfile", this.dockerFilePath);
+      copyFile(stagingDirectory, "app.yaml", this.appYamlPath);
+      copyFile(stagingDirectory, "Dockerfile", this.dockerFilePath);
+    } catch (IOException e) {
+      logger.error(e);
+      callback.errorOccurred(GctBundle.message("appengine.deployment.error.during.staging"));
```

Comment:
Yeah. Thanks for pointing them out. I'd changed these but not pushed them
yet.

On Tue, Mar 22, 2016 at 3:32 PM, Etan Shaul <notifications@github.com>
wrote:

> In
> google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/DoManagedVmDeployment.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/518#discussion_r57053481>
> :
>
> >
> > -    copyFile(stagingDirectory, "app.yaml", this.appYamlPath);
> > -    copyFile(stagingDirectory, "Dockerfile", this.dockerFilePath);
> > +      copyFile(stagingDirectory, "app.yaml", this.appYamlPath);
> > +      copyFile(stagingDirectory, "Dockerfile", this.dockerFilePath);
> > +    } catch (IOException e) {
> > +      logger.error(e);
> > +      callback.errorOccurred(GctBundle.message("appengine.deployment.error.during.staging"));
>
> looks like there are a few more missing returns in the class. I assume
> that after each callback.errorOccurred .. we want to return immediately.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/518/files/5e1503904678e7beb0ab336b0b01817738c60911#r57053481>
>